### PR TITLE
[updates] set up case updates

### PR DESCRIPTION
**Pull request by stephaniewong2**
### Description
[//]: # "Required - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."
Users can view updates per case by going into the Case screen and clicking "View Updates" below `CaseStatus`. This loads a list of updates sorted in chronological order. Each update contains the title, date, and blurb. Clicking into an update leads to the update screen which should also contain the update summary. No styling has been added yet. 

### Screenshots
[//]: # "Required for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy! Use the provided image template. Drag the desired image into the PR, then copy the link into the placeholder."

## How to review
[//]: # 'Required - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?'
- Add some updates with all fields to Supabase `updates` table.
- Click into the case that has updates and click "View Updates"
- Click into each update and confirm that the case date, title, blurb, and summary are present. 

## Next steps
[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."
- Style this flow.

### Online sources
[//]: # 'Optional - copy links to any tutorials or documentation that was useful to you when working on this PR'

### Related PRs
[//]: # "Optional - related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

CC: @ronniebeggs

### CODE REVIEW

File: src/app/(BottomTabNavigation)/AllCases/Updates/UpdateView/index.tsx [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+import styles from './styles';
+import { getUpdateById } from '../../../../../supabase/queries/updates';
+import { Update, UpdateUid } from '../../../../../types/types';
+import { formatDate } from '../../utils';
+
+export default function UpdateView() {
+  const { updateUid, lawFirm } = useLocalSearchParams<{
+    updateUid: UpdateUid;
+    lawFirm: string;
+  }>();
```
Comment (ronniebeggs) [inline src/app/(BottomTabNavigation)/AllCases/Updates/UpdateView/index.tsx]: Can we update this file to take the `updateUid` as the search parameter through the name of the file? We should also add the `lawFirm` to the update row in the supabase table. This should make integrating notifications and deeplinking easier.

File: src/app/(BottomTabNavigation)/AllCases/Updates/[caseUid].tsx [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        <View style={styles.updatesContainer}>
+          <FlatList
+            data={updates}
+            keyExtractor={item => item.updateUid}
+            ItemSeparatorComponent={() => <View style={styles.lineStyle} />}
+            renderItem={({ item }) => <UpdateItem {...item} />}
+            showsVerticalScrollIndicator={false}
+          />
+        </View>
+      </View>
```
Comment (ronniebeggs) [inline src/app/(BottomTabNavigation)/AllCases/Updates/[caseUid].tsx]: Can we render a temporarily "Loading..." text item in case no updates are associated with that case. We don't have this designed but we should include something temporary.

File: src/app/(BottomTabNavigation)/AllCases/_layout.tsx [hunk 3]
```diff
@@ -55,6 +55,18 @@ export default function CasesLayout() {
...
           }}
         />
+        <Stack.Screen
+          name="Updates/[caseUid]"
+          options={{
+            headerTitle: props => <CasesHeader />,
+          }}
+        />
+        <Stack.Screen
+          name="Updates/UpdateView/index"
```
Comment (ronniebeggs) [inline src/app/(BottomTabNavigation)/AllCases/_layout.tsx]: Change to `Updates/UpdateView/[updateUid]`.

File: src/supabase/queries/updates.ts [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+    updateUid: item.updateId,
+    caseUid: item.caseUUID,
+    title: item.title,
+    blurb: item.blurb,
+    category: item.category,
+    date: item.date,
+    summary: item.summary,
+  };
+  return updateData;
+}
```
Comment (ronniebeggs) [inline src/supabase/queries/updates.ts]: Can this be made so it's not an async function?

**Review by ronniebeggs — CHANGES_REQUESTED** [review]
I didn't really review style but everything else looks good so far. Hopefully we can clean everything up and make it more robust in future PRs. Good job!
